src/main/java/org/truffleruby/core/hash/HashNodes.java

```diff
@@ -890,6 +885,7 @@ protected RubyHash rehashBuckets(RubyHash hash,
             Arrays.fill(entries, null);
 
             Entry entry = hash.firstInSequence;
+            Entry previousEntry = null;
```

Comment:
`previousEntry` typically means the previous Entry in lookup, not in sequence, could you clarify by renaming it to `previousInSequence`?